common/service/pldm/pldm_firmware_update.c

```diff
@@ -622,7 +674,9 @@ static uint8_t update_component(void *mctp_inst, uint8_t *buf, uint16_t len, uin
 		goto exit;
 	}
 
-	if (current_state == STATE_IDLE) {
+	if (current_state != STATE_RDY_XFER) {
+		LOG_ERR("Current mode %d doesn't meet request state %d", current_state,
```

Comment:
Please update message to be more clear...something like "firmware update failed because current state %d is not in update mode"